casper/src/test/scala/coop/rchain/casper/api/BlockQueryResponseAPITest.scala

```diff
@@ -29,192 +27,176 @@ class BlockQueryResponseAPITest
-              )
-              b.bonds should be(secondBlock.bonds.map(BlockApi.bondToBondInfo))
-              b.blockSize should be(secondBlock.toProto.serializedSize.toString)
-              b.deployCount should be(secondBlock.state.deploys.length)
-              b.justifications should be(secondBlock.justifications.map(_.toHexString))
-          }
-        } yield ()
+  "getBlock" should "return successful block info response" in {
+    implicit val (blockStore, blockDagStorage, runtimeManager) = Mocks.create[Task]()
+
+    for {
+      _                  <- prepareDagStorage[Task]
+      blockApi           <- createBlockApi[Task]("", 1)
+      hash               = secondBlock.blockHash.toHexString
+      blockQueryResponse <- blockApi.getBlock(hash)
+      _ = inside(blockQueryResponse) {
+        case Right(blockInfo) =>
+          blockInfo.deploys should be(
+            randomDeploys.map(_.toDeployInfo)
+          )
+          val b = blockInfo.blockInfo
+          b.blockHash should be(secondBlock.blockHash.toHexString)
+          b.sender should be(secondBlock.sender.toHexString)
+          b.blockSize should be(secondBlock.toProto.serializedSize.toString)
+          b.seqNum should be(secondBlock.toProto.seqNum)
+          b.sig should be(secondBlock.sig.toHexString)
+          b.sigAlgorithm should be(secondBlock.sigAlgorithm)
+          b.shardId should be(secondBlock.toProto.shardId)
+          b.version should be(secondBlock.version)
+          b.blockNumber should be(secondBlock.blockNumber)
+          b.preStateHash should be(
+            secondBlock.preStateHash.toHexString
+          )
+          b.postStateHash should be(
+            secondBlock.postStateHash.toHexString
+          )
+          b.bonds should be(secondBlock.bonds.map(BlockApi.bondToBondInfo))
+          b.blockSize should be(secondBlock.toProto.serializedSize.toString)
+          b.deployCount should be(secondBlock.state.deploys.length)
+          b.justifications should be(secondBlock.justifications.map(_.toHexString))
       }
+    } yield ()
   }
 
-  it should "return error when no block exists" in withStorage {
-    implicit blockStore => implicit blockDagStorage =>
-      runtimeManagerResource.use { implicit runtimeManager =>
-        for {
-          blockApi           <- createBlockApi[Task]("", 1)
-          hash               = badTestHashQuery
-          blockQueryResponse <- blockApi.getBlock(hash)
-          _ = inside(blockQueryResponse) {
-            case Left(msg) =>
-              msg should be(
-                s"Error: Failure to find block with hash: $badTestHashQuery"
-              )
-          }
-        } yield ()
+  it should "return error when no block exists" in {
+    implicit val (blockStore, blockDagStorage, runtimeManager) = Mocks.create[Task]()
+
+    for {
+      blockApi           <- createBlockApi[Task]("", 1)
+      hash               = badTestHashQuery
+      blockQueryResponse <- blockApi.getBlock(hash)
```

Comment:
Better not to replicate the existing tests which will just make more confusion.

In these tests we just want to check is dependencies are called with correct data from the input in API, and also to check that any result received from dependencies is correctly returned to API methods.
The meaning of this data is not concern of the API, except if is part of validation of the input.